code/obj/flock/structure/rift.dm

```diff
@@ -27,34 +27,35 @@
 	src.info_tag.set_info_tag("Entry time: [round(src.build_time - elapsed)] seconds")
 	if(elapsed >= build_time)
 		src.visible_message("<span class='text-blue'>Multiple shapes exit out of [src]!</span>")
-		for(var/i in 1 to pick(3, 4))
+		for(var/i in 1 to 4)
 			var/obj/item/flockcache/x = new(src.contents)
-			x.resources = rand(40, 50)
+			x.resources = 40
 			eject += x
 		for(var/i in 1 to 4)
 			var/obj/flock_structure/egg/e = new(src.contents, src.flock)
 			eject += e
 		var/obj/flock_structure/egg/bit/bitegg = new(src.contents, src.flock)
 		eject += bitegg
 		var/list/candidate_turfs = list()
-		for(var/turf/simulated/floor/S in orange(src, 4))
+		for(var/turf/simulated/floor/S in range(src, 2)) // 5x5
+			if (istype(S, /turf/simulated/floor/feather))
+				continue
 			candidate_turfs += S
+		while (length(candidate_turfs) > 13)
+			candidate_turfs -= pick(candidate_turfs)
```

Comment:
Considering you already shuffle the turfs you could just move the shuffling here and truncate it to length 13 and get the same effect but faster and cleaner. But the numbers are so small that speed shouldn't matter much.